src/mewbot/api/registry.py

```diff
@@ -121,14 +165,48 @@ def do_register(api: Type[Component]) -> Type[Component]:
 
     @classmethod
     def _detect_api_versions(mcs, impl: Type[Any]) -> Iterable[Tuple[ComponentKind, str]]:
+        """Finds all API versions that match a given implementation."""
+
         for kind, apis in mcs._api_versions.items():
             for version, api in apis.items():
                 if issubclass(impl, api):
                     yield kind, version
 
     @classmethod
-    def api_version(mcs, component: Component) -> Tuple[ComponentKind, str]:
-        for val in mcs._detect_api_versions(type(component)):
+    def api_version(mcs, component: Component | Type[Component]) -> Tuple[ComponentKind, str]:
+        """
+        Gets the Component Kind (e.g. 'Behaviour') and API version (e.g. 'v1') from a component.
+
+        This function will only return the first matched type, as per the Registry's
+        'one base type' constraint.
+        """
+
+        if not isinstance(component, type):
+            component = type(component)
+
+        for val in mcs._detect_api_versions(component):
             return val
 
```

Comment:
Suggested change - as far as I can tell, for well designed code which is behaving how we'd prefer, this should never trip more than once (if nothing else, there's a check elsewhere which makes it unlikely).

Thus I propose it should error, instead of returning the first matched value, if there is more than one match. Seems less likely to produce unexpected behavior and a safer behavior. People could still work around it, but at least they would know they where doing something a bit didgy. 

(But there might well be some use case I'm missing which would be a shame to loose, so I'm not married to the idea).

Reply:
Hmm. Yeah - it _should_ never occur unless, as you say, someone has messed with subclass hooks.

It just seems like the sort of assumption that might trip someone up at some point (it should match once and only once, but, currently, it silently works if, afact, there's more than one match).

It's probably not an issue! I'd only suggest closing it off out of an abundance of caution.

Reply:
My opinion would be if it's a potential error state - even if we don't know how the error could occur - it should report an error; but then that's just in the effort of robustness not efficiency.

Reply:
754854a has been included in the tree